### PR TITLE
V2.5 update machine controller

### LINKED ISSUE #1194: [Arvato] Fix loading of Openstack region
The creation of a machines fails with:
```
E0606 16:14:58.233811       1 machine.go:213] machine-kubermatic-5hr8w5jv87-x2w44 failed with: invalid provider config: failed to get region "regionOne": Resource not found
I0606 16:14:58.233929       1 main.go:386] Event(v1.ObjectReference{Kind:"Machine", Namespace:"", Name:"machine-kubermatic-5hr8w5jv87-x2w44", UID:"6b3247c1-699e-11e8-8d4e-0a580a9e0335", APIVersion:"machine.k8s.io", ResourceVersion:"4890", FieldPath:""}): type: 'Warning' reason: 'ValidationFailed' Validation failed: failed to get region "regionOne": Resource not found
```

Though listing all regions with the same credentials results in:
```
bom03@devopsw2:~/CMP/terraform-openstack/bom03/TEN0001031-OSP-ApplServicesOSP$ openstack region show regionOne
+---------------+-----------+
| Field         | Value     |
+---------------+-----------+
| description   |           |
| parent_region | None      |
| region        | regionOne |
+---------------+-----------+
```

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
V2.5 update machine controller to fix the openstack endpoint availability issue

Fixes #1194

### CODE REVIEW

File: api/pkg/handler/authentication.go [hunk 1]
```diff
@@ -45,9 +47,20 @@ type openIDAuthenticator struct {
 func NewOpenIDAuthenticator(issuer, clientID string, extractor TokenExtractor, insecureSkipVerify bool) (Authenticator, error) {
 	ctx := context.Background()
 	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: insecureSkipVerify},
+		Proxy: http.ProxyFromEnvironment,
```
Comment (alvaroaleman) [inline api/pkg/handler/authentication.go]: What happens if there is no proxy defined in the env?

File: api/pkg/resources/apiserver/deployment.go [hunk 2]
```diff
@@ -276,6 +282,12 @@ func getApiserverFlags(data *resources.TemplateData, externalNodePort int32) []s
 		flags = append(flags, "--cloud-provider", "openstack")
 		flags = append(flags, "--cloud-config", "/etc/kubernetes/cloud/config")
 	}
+
+	if data.Cluster.Spec.Cloud.BringYourOwn != nil {
```
Comment (alvaroaleman) [inline api/pkg/resources/apiserver/deployment.go]: Uhm. How can this be determined for sure and everyone by just taking the cluster type in account? BYOWN may well not have routable node names. Maybe just make this a config option?

**Review by alvaroaleman — CHANGES_REQUESTED** [review]

**Comment by mrIncompetent** [comment]
Sorry wrong branch :man_facepalming:

**Comment by mrIncompetent** [comment]
@alvaroaleman changed target branch  to v2.5. Was master before

File: api/pkg/handler/authentication.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/handler/authentication.go]: Though not relevant atm:
If not defined, no proxy gets used.

**Review by alvaroaleman — APPROVED** [review]

File: api/pkg/resources/apiserver/deployment.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/resources/apiserver/deployment.go]: Was a temporary workaround for a issue we had at Arvato.
When finally moving apiserver->kubelet communication over to the VPN, we have a consistent way of handling it. Until then, we won't touch it i guess. We're using BYO exclusively at Arvato

**Pull request merged**